### PR TITLE
Add approver and label to addon-manager

### PR DESCRIPTION
**Pull request by MrHohn**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Adding myself as an approver might make rebasing image and bumping version easier (e.g. https://github.com/kubernetes/kubernetes/pull/76404, https://github.com/kubernetes/kubernetes/pull/76405). (Not necessity but good to have :))

/assign @mikedanese 

**Which issue(s) this PR fixes**:

Fixes #NONE 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cluster/addons/addon-manager/OWNERS [hunk 1]
```diff
@@ -2,3 +2,7 @@
 
 reviewers:
 - mrhohn
+approvers:
+- mrhohn
+labels:
+- sig/cluster-lifecycle
```
Comment (mikedanese) [inline cluster/addons/addon-manager/OWNERS]: Is this used anywhere other than GCP? Does cluster-lifecycle actually want to own this?

**Review by mikedanese — COMMENTED** [review]
lgtm, one question

File: cluster/addons/addon-manager/OWNERS (hunk 1 above)
Comment (MrHohn) [reply cluster/addons/addon-manager/OWNERS]: Good point, updated this to `sig/gcp`.

**Comment by MrHohn** [comment]
@mikedanese Would also need approve label here, thanks!

**Pull request merged**